tzatziki-spring-mongodb/src/main/java/com/decathlon/tzatziki/steps/SpringMongoSteps.java

```diff
@@ -46,6 +51,51 @@ public SpringMongoSteps(ObjectSteps objects, SpringSteps spring) {
 
     @Before
     public void before() {
+        if (autoclean) {
+            mongoTemplates.forEach(template -> template.getDb().drop());
+        }
+
+        if (mongoRepositoryByClass == null) {
+            mongoRepositoryByClass = spring.applicationContext().getBeansOfType(MongoRepository.class).values()
+                    .stream()
+                    .map(mongoRepository -> Map.entry(mongoRepository, TypeUtils.getTypeArguments(mongoRepository.getClass(), MongoRepository.class).get(MongoRepository.class.getTypeParameters()[0])))
+                    .sorted((e1, e2) -> {
+                        if (e1.getValue() instanceof Class) return -1;
+                        return e2.getValue() instanceof Class ? 1 : 0;
+                    })
+                    .<Map.Entry<Class<?>, MongoRepository<?, ?>>>mapMulti((mongoRepositoryWithType, consumer) -> {
+                        MongoRepository<?, ?> mongoRepository = mongoRepositoryWithType.getKey();
+                        Type type = mongoRepositoryWithType.getValue();
+                        if (type instanceof TypeVariable<?> typeVariable) {
+                            type = typeVariable.getBounds()[0];
+                            TypeParser.getSubtypesOf((Class<?>) type)
+                                    .forEach(clazz -> consumer.accept(Map.entry(clazz, mongoRepository)));
+                        }
+
+                        if (type instanceof Class<?> clazz) consumer.accept(Map.entry(clazz, mongoRepository));
+                    })
+                    .collect(Collectors.toMap(
+                            Map.Entry::getKey,
+                            Map.Entry::getValue,
+                            (v1, v2) -> v1
+                    ));
+        }
+        if (entityClassByCollectionName == null) {
+            entityClassByCollectionName = mongoRepositoryByClass.keySet().stream()
+                    .map(type -> (Class<?>) type)
```

Comment:
I don't understand the purpose of those lines of code. In my opinion, only
`        if (autoclean) {
            mongoTemplates.forEach(template -> template.getDb().drop());
        }`  is enough for the feature.

Reply:
Yep, i don't remember why... it's fixed